From c6b065a301a5fddbbb51aa37fbe015731d2af232 Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Fri, 4 Nov 2011 20:48:36 +0100
Subject: [PATCH 158/327] sched-clear-pf-thread-bound-on-fallback-rq.patch

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 kernel/sched/core.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index 123d2ec6520f..acb13a9b6bb1 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -1247,6 +1247,12 @@ out:
 		}
 	}
 
+	/*
+	 * Clear PF_NO_SETAFFINITY, otherwise we wreckage
+	 * migrate_disable/enable. See optimization for
+	 * PF_NO_SETAFFINITY tasks there.
+	 */
+	p->flags &= ~PF_NO_SETAFFINITY;
 	return dest_cpu;
 }
 
-- 
2.1.1

