From 798f4591d0d352b70db8215188133217f550a1b4 Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Mon, 14 Nov 2011 10:57:54 +0100
Subject: [PATCH] rcu-more-fallout.patch

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>

diff --git a/kernel/rcutiny.c b/kernel/rcutiny.c
index 9ed6075..f202b26 100644
--- a/kernel/rcutiny.c
+++ b/kernel/rcutiny.c
@@ -369,6 +369,7 @@ void call_rcu_sched(struct rcu_head *head, void (*func)(struct rcu_head *rcu))
 }
 EXPORT_SYMBOL_GPL(call_rcu_sched);
 
+#ifndef CONFIG_PREEMPT_RT_FULL
 /*
  * Post an RCU bottom-half callback to be invoked after any subsequent
  * quiescent state.
@@ -378,6 +379,7 @@ void call_rcu_bh(struct rcu_head *head, void (*func)(struct rcu_head *rcu))
 	__call_rcu(head, func, &rcu_bh_ctrlblk);
 }
 EXPORT_SYMBOL_GPL(call_rcu_bh);
+#endif
 
 void rcu_init(void)
 {
-- 
2.1.0

