From 93c5045786fa42629c93dd135b1bb83e10123581 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (Red Hat)" <rostedt@goodmis.org>
Date: Tue, 8 Jul 2014 18:37:31 -0400
Subject: [PATCH 326/326] Linux 3.12.22-rt35 REBASE

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index c5b71f9a229d..366440d74b77 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt25
+-rt35
-- 
2.0.0

